File: backend/db/run_migration.py
import os
import psycopg2
from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT

# Database connection parameters
db_params = {
    "host": os.environ.get("DB_HOST", "postgres"),
    "database": os.environ.get("DB_NAME", "spheroseg"),
    "user": os.environ.get("DB_USER", "spheroseg"),
    "password": os.environ.get("DB_PASSWORD", "spheroseg"),
}

def run_migration():
    """Run the database migration script"""
    print("Starting database migration...")
    
    # Connect to the database
    connection = psycopg2.connect(**db_params)
    connection.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
    cursor = connection.cursor()
    
    try:
        # Read the migration SQL file
        with open("./db/migration.sql", "r") as f:
            migration_sql = f.read()
        
        # Execute the migration script
        print("Executing migration SQL...")
        cursor.execute(migration_sql)
        
        print("Migration completed successfully!")
    except Exception as e:
        print(f"Error during migration: {e}")
    finally:
        cursor.close()
        connection.close()

if __name__ == "__main__":
    run_migration()

File: packages/backend/src/scripts/run-migration.js
/**
 * Simple migration script for access_requests table
 */

const { Pool } = require('pg');
const fs = require('fs');
const path = require('path');

// Database configuration
const pool = new Pool({
  host: process.env.DB_HOST || 'localhost',
  port: parseInt(process.env.DB_PORT || '5432'),
  database: process.env.DB_NAME || 'spheroseg',
  user: process.env.DB_USER || 'postgres',
  password: process.env.DB_PASSWORD || 'postgres',
});

async function runMigration() {
  try {
    console.log('🔄 Running access_requests table migration...');
    
    // Read the migration file
    const migrationPath = path.join(__dirname, '../db/migrations/004_create_access_requests.sql');
    const migrationSQL = fs.readFileSync(migrationPath, 'utf8');
    
    // Execute the migration
    await pool.query(migrationSQL);
    
    console.log('✅ Migration completed successfully!');
    console.log('📋 access_requests table is now available');
    
  } catch (error) {
    console.error('❌ Migration failed:', error.message);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

// Run if called directly
if (require.main === module) {
  runMigration();
}

module.exports = { runMigration };